code/modules/overmap/ships/computers/shuttle.dm

```diff
@@ -4,25 +4,9 @@
 	ui_template = "shuttle_control_console_exploration.tmpl"
 	base_type = /obj/machinery/computer/shuttle_control/explore
 
-	var/eye_type = /mob/observer/eye/landing/
-	var/mob/observer/eye/landing/eyeobj = null
-
-	// Manual landing locations.
-	var/datum/action/shuttle/finish_landing/finish_landing_action
-	var/datum/action/shuttle/end_landing/end_landing_action
-	var/mob/current_user
-
 /obj/machinery/computer/shuttle_control/explore/Initialize()
 	. = ..()
-	finish_landing_action = new(src)
-	end_landing_action = new(src)
-
-/obj/machinery/computer/shuttle_control/explore/Destroy()
-	end_landing()
-	current_user = null
-	QDEL_NULL(finish_landing_action)
-	QDEL_NULL(end_landing_action)
-	. = ..()
+	set_extension(src, /datum/extension/eye/landing)
```

Comment:
What happens to the AIs vision if it ever uses this computer?

Reply:
Currently, nothing. The look() proc on the extension returns if the user already has an eyeobj. Caching the old eyeobj instead may be possible.